pkg/pool-gyro/contracts/lib/Gyro2CLPMath.sol

```diff
@@ -116,17 +116,26 @@ library Gyro2CLPMath {
         uint256 a,
         uint256 mb,
         uint256 bSquare, // b^2 can be calculated separately with more precision
-        uint256 mc
+        uint256 mc,
+        Rounding rounding
     ) internal pure returns (uint256 invariant) {
-        uint256 denominator = a.mulUp(2 * FixedPoint.ONE);
+        function(uint256, uint256) pure returns (uint256) _mulUpOrDown = rounding == Rounding.ROUND_DOWN
+            ? FixedPoint.mulDown
+            : FixedPoint.mulUp;
+
+        function(uint256, uint256) pure returns (uint256) _divUpOrDown = rounding == Rounding.ROUND_DOWN
+            ? FixedPoint.divDown
+            : FixedPoint.divUp;
+
+        uint256 denominator = 2 * a;
```

Comment:
ah nice. This result is exact so it doesn't need rounding right?

Reply:
Exactly

Reply:
Besides, solc 0.8.24 handles overflows and underflows. I believe that was the reason to use mulDown for exact operations in the old library.